pkg/query-service/agentConf/manager.go

```diff
@@ -0,0 +1,217 @@
+	return m.GetConfigHistory(ctx, typ)
+}
+
+// StartNewVersion launches a new config version for given set of elements
+func StartNewVersion(ctx context.Context, eleType ElementTypeDef, elementIds []string) (*ConfigVersion, error) {
+
+	if !m.Ready() {
+		// agent is already being updated, ask caller to wait and re-try after sometime
+		return nil, fmt.Errorf("agent updater is busy")
+	}
+
+	// create a new version
+	cfg := NewConfigversion(eleType)
+
+	// insert new config and elements into database
+	err := m.insertConfig(ctx, cfg, elementIds)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func Redeploy(ctx context.Context, typ ElementTypeDef, version int) error {
+	if !atomic.CompareAndSwapUint32(&m.lock, 0, 1) {
+		return fmt.Errorf("agent updater is busy")
+	}
+
+	defer atomic.StoreUint32(&m.lock, 0)
+
+	configVersion, err := GetConfigVersion(ctx, typ, version)
+	if err != nil {
+		zap.S().Debugf("failed to fetch config version during redeploy", err)
+		return fmt.Errorf("failed to fetch details of the config version")
+	}
+
+	if configVersion == nil || (configVersion != nil && configVersion.LastConf == "") {
+		zap.S().Debugf("config version has no conf yaml", configVersion)
+		return fmt.Errorf("the config version can not be redeployed")
+	}
+	switch typ {
+	case ElementTypeSamplingRules:
+		var config *tsp.Config
+		if err := yaml.Unmarshal([]byte(configVersion.LastConf), &config); err != nil {
+			zap.S().Errorf("failed to read last conf correctly", err)
+			return fmt.Errorf("failed to read the stored config correctly")
+		}
+
+		// merge current config with new filter params
+		processorConf := map[string]interface{}{
+			"signoz_tail_sampling": config,
+		}
+
+		opamp.AddToTracePipelineSpec("signoz_tail_sampling")
+		configHash, err := opamp.UpsertControlProcessors(ctx, "traces", processorConf, m.OnConfigUpdate)
+		if err != nil {
+			zap.S().Errorf("failed to call agent config update for trace processor:", err)
+			return fmt.Errorf("failed to deploy the config")
+		}
+
+		m.updateDeployStatus(ctx, ElementTypeSamplingRules, version, string(DeployInitiated), "Deployment started", configHash, configVersion.LastConf)
+	case ElementTypeDropRules:
+		var filterConfig *filterprocessor.Config
+		if err := yaml.Unmarshal([]byte(configVersion.LastConf), &filterConfig); err != nil {
+			zap.S().Errorf("failed to read last conf correctly", err)
+			return fmt.Errorf("failed to read the stored config correctly")
+		}
+		processorConf := map[string]interface{}{
+			"filter": filterConfig,
+		}
+
+		opamp.AddToMetricsPipelineSpec("filter")
+		configHash, err := opamp.UpsertControlProcessors(ctx, "metrics", processorConf, m.OnConfigUpdate)
+		if err != nil {
+			zap.S().Errorf("failed to call agent config update for trace processor:", err)
+			return err
+		}
+
+		m.updateDeployStatus(ctx, ElementTypeSamplingRules, version, string(DeployInitiated), "Deployment started", configHash, configVersion.LastConf)
```

Comment:
Same question here.

Reply:
not sure. what is the question? this method updates status of version.
The version is created upfront before we push updates.